File: DEMO_VanHungThinh/Web_VanHungThinh_Demo/API_VanThinhPhat/Models/Delivery.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace API_VanHungThinh.Models
{
    public class Delivery
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("billId")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string BillId { get; set; }

        [BsonElement("deliveryDate")]
        public DateTime DeliveryDate { get; set; }

        [BsonElement("deliveredBy")]
        public string DeliveredBy { get; set; }

        [BsonElement("recipient")]
        public string Recipient { get; set; }

        [BsonElement("status")]
        public string Status { get; set; }

        [BsonElement("notes")]
        public string Notes { get; set; }
    }
}

File: frontend-sample/packages/react/src/types/delivery.ts
export type DeliveryStatus = "Đang chờ" | "Đã giao" | "Đã hủy";

export interface Delivery {
  id?: string;
  billId: string;
  deliveryDate: Date;
  deliveredBy: string;
  recipient: string;
  status: string;
  notes?: string;
}


export interface CreateDeliveryRequest {
  billId: string;
  deliveryDate: Date;
  deliveredBy: string;
  recipient: string;
  status: DeliveryStatus;
  notes?: string;
}
